File: src/dataset/cpp/raindrops_Exercism_cmccandless/claude/ClaudeSonnet4_raindrops_v1.cpp
#include "raindrops.h"

string raindrops::convert(int x)
{
  string result;
  
  bool has_factor = false;
  
  if (x % 3 == 0) {
     result += "Pling";
     has_factor = true;
  }
  if (x % 5 == 0) {
     result += "Plang";
     has_factor = true;
  }
  if (x % 7 == 0) {
     result += "Plong";
     has_factor = true;
  }
  
  if (!has_factor) {
     result = std::to_string(x);
  }
  
  return result;
}

File: src/dataset/typescript/raindrops_Exercism-typescript-shybyte/claude/ClaudeSonnet4_raindrops_v1.ts
const FACTOR_RAINDROP_PAIRS: ReadonlyArray<readonly [number, string]> = [
    [3, 'Pling'],
    [5, 'Plang'],
    [7, 'Plong']
] as const

export default class Raindrops {
    convert(n: number): string {
        let result = ''
        
        for (const [factor, drop] of FACTOR_RAINDROP_PAIRS) {
            if (n % factor === 0) {
                result += drop
            }
        }
        
        return result || n.toString()
    }
}